CMakeLists.txt

```diff
@@ -110,6 +110,14 @@ if(CCACHE_FOUND)
     message("-- Found ccache: ${CCACHE_FOUND}")
 endif(CCACHE_FOUND)
 
+if (UNIX AND NOT APPLE)
+  execute_process(COMMAND ${CMAKE_C_COMPILER} -fuse-ld=gold -Wl,--version ERROR_QUIET OUTPUT_VARIABLE ld_version)
+  if ("${ld_version}" MATCHES "GNU gold")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=gold -Wl,--disable-new-dtags")
```

Comment:
I assume we used bfd so far. Switching linkers could potentially cause unexpected behaviours and serious errors. Should we make gold the default? Are we sure this works properly on all supported platforms?

Reply:
That's actually a very good point, this was a branch with a ton of commits where I attempted various things. I'd like to keep this with a conditional.

Reply:
I am completely for keeping it with a conditional.